### DISCUSSION TITLE
Azure Data Factory CI/CD with GitHub Actions giving and error

### DISCUSSION THREAD

**Discussion by hasithamgh** [community/community#36472] [Actions]
I’m working on an Azure Data Factory deployment using GitHub actions. [Here](https://techcommunity.microsoft.com/t5/fasttrack-for-azure/azure-data-factory-ci-cd-with-github-actions/bc-p/3652967#M269) is the article I followed on creating the action.
However, I'm getting the following error while executing the workflow. 
Highly appreciate your assistance on resolving this as I’m stuck at this point.
 
**Error message:** 
=====================================================
Run Azure/arm-deploy@v1.0.[6](https://github.com/hasithaorg/AzureDataFactoryDemo/actions/runs/3245211824/jobs/5322444299#step:4:6)
  with:
    scope: resourcegroup
    resourceGroupName: ***
    template: ADFSKDemo/ARMTemplateForFactory.json
    deploymentMode: Incremental
    parameters: factoryName=ADFSKDemo01
    failOnStdErr: true
  env:
    AZURE_HTTP_USER_AGENT: 
    AZUREPS_HOST_ENVIRONMENT: 
Validating template...
Warning: ERROR: 'str' object has no attribute 'get'

Warning: Template validation failed.
Creating deployment...
Error: ERROR: 'str' object has no attribute 'get'

Error: Deployment failed.
 

**Detailed error after enabling debugging mode:**

=================================================
##[debug]Evaluating condition for step: 'Deploy Azure Resource Manager (ARM) Template'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Deploy Azure Resource Manager (ARM) Template
##[debug]Loading inputs
##[debug]Evaluating: secrets.AZURE_RG
##[debug]Evaluating Index:
##[debug]..Evaluating secrets:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'AZURE_RG'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Loading env
Run Azure/arm-deploy@v[1](https://github.com/hasithaorg/AzureDataFactoryDemo/actions/runs/3244732711/jobs/5321292388#step:4:1).0.[6](https://github.com/hasithaorg/AzureDataFactoryDemo/actions/runs/3244732711/jobs/5321292388#step:4:6)
Validating template...
Warning: ERROR: 'str' object has no attribute 'get'

Warning: Template validation failed.
Creating deployment...
##[debug]exec tool: /usr/bin/az
##[debug]arguments:
##[debug]   deployment
##[debug]   group
##[debug]   create
##[debug]   --resource-group
##[debug]   ***
##[debug]   --template-file
##[debug]   ./ADFSKDemo/ARMTemplateForFactory.json
##[debug]   --mode
##[debug]   incremental
##[debug]   --parameters
##[debug]   factoryName=ADFSKDemo01
##[debug]   -o
##[debug]   json
Error: ERROR: 'str' object has no attribute 'get'

##[debug]Exit code 1 received from tool '/usr/bin/az'
##[debug]STDIO streams have closed for tool '/usr/bin/az'
Error: Deployment failed.
##[debug]Node Action run completed with exit code 1
##[debug]Finishing: Deploy Azure Resource Manager (ARM) Template
 
thank you!

**Comment by ossanna16** [comment]
Hey there @hasithamgh and thank you for asking a good question! I'm glad you found our community 🙂

**Reply by hasithamgh** [reply]
Sure @ossanna16, there isn't a lot of resources for GitHub actions than this community I guess 🙂

**Reply by ossanna16** [reply]
Hey there @hasithamgh! What kind of resources are you looking for specifically?
